BlueprintUI/Sources/Measuring/TypeKeyedCache.swift

```diff
@@ -0,0 +1,34 @@
+import UIKit
+
+/// A cache that uses the value's type as the key.
+public final class TypeKeyedCache {
+
+    private var views: [Key: AnyObject] = [:]
+
+    // Intentionally internal. Not intended for public instantiation.
+    init() {}
+
+    // Returns the cached value for the Value type if it exists. If it doesn't
+    // exist, it creates a new instance of Value, caches it, and returns it.
+    //
+    // - Parameter create: A closure that creates a new instance of Value. It
+    // is only invoked when a cached value does not exist. The created value is
+    // cached for future usage.
+    public func value<Value: AnyObject>(_ create: () -> Value) -> Value {
```

Comment:
FWIW, I went with the "access" pattern with an access closure so people aren't tempted to keep a reference to the cached measurement view, kind of like the swift pointer APIs where the pointer shouldn't be stored outside the access closure.